vmm/src/vm.rs

```diff
@@ -1241,7 +1241,8 @@ impl Vm {
         let guest_memory = memory_manager.lock().as_ref().unwrap().guest_memory();
         let mem = guest_memory.memory();
         let alignment = 0x20_0000;
-        let aligned_kernel_addr = arch::layout::KERNEL_START.0 + (alignment - 1) & !(alignment - 1);
+        let aligned_kernel_addr =
```

Comment:
may be a bit hard to read in github UI: added additional `()`

Reply:
Are you sure about that change - here is an existing function to do it:

```
#[inline]
fn align_down<T>(val: T, align: T) -> T
where
    T: BitAnd<Output = T> + Not<Output = T> + Sub<Output = T> + From<u8>,
{
    val & !(align - 1u8.into())
}
```

Reply:
I'm not sure why you just wouldn't make the load address aligned in the first place - it's a constant in the source? @RuoqingHe

Reply:
For now shall we just replace with `aligned_kernel_addr = arch::layout::KERNEL_START.0.div_ceil(alignment) * alignment` ? Which is a nice clean way to align upwards.

Reply:
SGTM